src/pymoca/backends/casadi/api.py

```diff
@@ -191,6 +191,45 @@ def save_model(model_folder: str, model_name: str, model: Model,
         for key in ['states', 'der_states', 'alg_states', 'inputs', 'parameters', 'constants']:
             db[key] = [e.to_dict() for e in getattr(model, key)]
 
+
+        # Caching using CasADi functions will lead to constants seemingly
+        # depending on MX variables. Figuring out that they do not is slow,
+        # especially when doing it on a lazy function call, as would be the
+        # case when reading from cache. So instead, we do the depency check
+        # once when saving the model.
+
+        # Metadata dependency checking
+        parameter_vector = ca.veccat(*[v.symbol for v in model.parameters])
+
+        for k, key in enumerate(['states', 'alg_states', 'inputs', 'parameters', 'constants']):
+            metadata_shape = (len(getattr(model, key)), len(CASADI_ATTRIBUTES))
+            m = db[key + "__metadata_dependent"] = np.zeros(metadata_shape, dtype=bool)
+            for i, v in enumerate(getattr(model, key)):
+                for j, tmp in enumerate(CASADI_ATTRIBUTES):
+                    attr = getattr(v, tmp)
+                    if isinstance(attr, ca.MX) and ca.depends_on(attr, parameter_vector):
+                        m[i, j] = True
+
+        # Delay dependency checking
+        if model.delay_states:
+
+            all_symbols = [model.time,
+                           *model._symbols(model.states),
+                           *model._symbols(model.der_states),
+                           *model._symbols(model.alg_states),
+                           *model._symbols(model.inputs),
+                           *model._symbols(model.constants),
+                           *model._symbols(model.parameters)]
+            all_hashes = [hash(x) for x in all_symbols]
```

Comment:
it is not immediately obvious why you are using a hash. I imagine it is faster to do a linear search on? Why not use an an ordered dict or something else with fast lookups?

Reply:
Something like
```python
{x: i for i, x in enumerate(all_symbols)}
```

Reply:
Fixed.